test/libsolidity/syntaxTests/inlineAssembly/shadowing/variable_by_opcode.sol

```diff
@@ -0,0 +1,14 @@
+contract C {
+    function f() public pure {
+        uint mload;
+    }
+    function g() public pure {
+        uint mload;
+        assembly {
+        }
+    }
+}
+// ----
+// Warning 8261: (109-119): Variable is shadowed in inline assembly by an instruction of the same name
```

Comment:
This check only executes if there is an assembly block, and it uses `evmasm::c_instructions` and not the dialect.